File: BankOfMyHouse.Server/BankOfMyHouse.API/Endpoints/Users/GetUserDetails/GetUserDetailsResponseDto.cs
namespace BankOfMyHouse.API.Endpoints.Users.GetUserDetails;

public sealed record GetUserDetailsResponseDto
{
	public int Id { get; set; }
	public string Username { get; set; } = string.Empty;
	public string Email { get; set; } = string.Empty;
	public string? FirstName { get; set; }
	public string? LastName { get; set; }
	public DateTimeOffset CreatedAt { get; set; }
	public DateTimeOffset? LastLoginAt { get; set; }
	public bool IsActive { get; set; }
	public List<string> Roles { get; set; } = new();
	public List<BankAccountDetailsDto> BankAccounts { get; set; } = new();
}

public sealed record BankAccountDetailsDto
{
	public int Id { get; set; }
	public string IBAN { get; set; } = string.Empty;
	public decimal Balance { get; set; }
	public DateTimeOffset CreationDate { get; set; }
}

File: BankOfMyHouse.Client/src/app/auth/models/getUserDetails/getUserDetailsResponseDto.ts
export class GetUserDetailsResponseDto {
    id: number = 0;
    username: string = '';
    email: string = '';
    firstName?: string;
    lastName?: string;
    createdAt: Date = new Date();
    lastLoginAt?: Date;
    isActive: boolean = false;
    roles: string[] = [];
    bankAccounts: BankAccountDetailsDto[] = [];
}

export class BankAccountDetailsDto {
    id: number = 0;
    iban: string = '';
    balance: number = 0;
    creationDate: Date = new Date();
}